### PR TITLE
render fix for tag color and write some unit tests

### PR DESCRIPTION
**Pull request by dmlb**
## Describe your changes
Adding some unit tests to confirm the colour utilities and component for tag colour are doing what they're supposed to
Fixes https://github.com/ClimateTown/knowledge-hub/issues/375
needed key on for loop!

### CODE REVIEW

File: src/lib/components/TagWrapper.test.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    )
+  })
+
+  it("does not add the tag-color class for colour not in the map", () => {
+    const tagComp = render(TagWrapper, { tagColor: "not-real" })
+    const divEl = tagComp.getByTestId("tag-wrapper-component")
+
+    expect(divEl.classList).not.toContain("tag-color")
+  })
+})
```
Comment (VeckoTheGecko) [inline src/lib/components/TagWrapper.test.ts:48]: We have testing for the data itself in [data/yaml_test.py](https://github.com/dmlb/knowledge-hub/blob/53adab5f9bf0ed178627f81b6dcc8dd39e807052/data/yaml_test.py) via pytest and some json schemas. I think some of these tests are doubling up here.

File: src/lib/components/FilterForm.svelte [hunk 2]
```diff
@@ -80,7 +80,7 @@
       class="p-4 space-y-4"
     >
       <div class="flex flex-row flex-wrap gap-2">
-        {#each filterOptions as filterOption}
+        {#each filterOptions as filterOption (filterOption)}
```
Comment (VeckoTheGecko) [inline src/lib/components/FilterForm.svelte:83]: Do you know whether this works with arbitrary data types? From what I've seen online they mainly use an int ID for the key. I have no idea here, so lmk what you think :P

**Review by VeckoTheGecko — COMMENTED** [review]
Thanks for the PR @dmlb ! See above for my feedback. I think end to end testing would be good for the project, but unit testing might be a bit overkill for the simplicity of the project. Let me know if you disagree though, curious to know what you think.

File: src/lib/components/TagWrapper.test.ts (hunk 1 above)
Comment (dmlb) [reply src/lib/components/TagWrapper.test.ts:48]: this are checking data validity, but this I was checking the HTML rendered the props i wanted in each case :)

File: src/lib/components/FilterForm.svelte (hunk 2 above)
Comment (dmlb) [reply src/lib/components/FilterForm.svelte:83]: from my testing you can stick anything in there, I think it transforms it all into a hash string, so an object would be most unique

File: src/lib/components/TagWrapper.test.ts (hunk 1 above)
Comment (VeckoTheGecko) [reply src/lib/components/TagWrapper.test.ts:48]: Ah, I see now

**Comment by VeckoTheGecko** [comment]
@dmlb I think this is ready to merge now?

In terms of workflow for PRs, feel free to re-request reviews so that I know that you're done making changes (or ping me with "good to go" or something).

I don't mind being pinged :)

**Comment by dmlb** [comment]
@VeckoTheGecko updated 👌

**Review by VeckoTheGecko — APPROVED** [review]
Looks good!

**Pull request merged**